engine/health_check.go

```diff
@@ -69,15 +71,43 @@ func (e *Engine) checkOneContainer(container *types.Container) {
 	if container.HealthCheck != nil {
 		timeout := time.Duration(e.config.HealthCheck.Timeout) * time.Second
 		container.Healthy = checkSingleContainerHealthy(container, timeout)
+		log.Debugf("[checkSingleContainerHealthy] check container %s health status: %v", container.ID, container.Healthy)
 	}
 
-	cctx, cancel := context.WithTimeout(context.Background(), e.config.GlobalConnectionTimeout)
-	defer cancel()
-	if err := e.store.SetContainerStatus(cctx, container, e.node, e.config.GetHealthCheckStatusTTL()); err != nil {
+	if err := e.setContainerStatus(container); err != nil {
 		log.Errorf("[checkOneContainer] update deploy status failed %v", err)
 	}
 }
 
+// 设置容器状态，允许重试，带timeout控制
+func (e *Engine) setContainerStatus(container *types.Container) error {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
```

Comment:
ha? I don't think you need this `cancel`...

Reply:
I notice the code below, since this `f()` has a context and will not run forever, maybe you don't need this `ctx` to control the cancellation of `BackoffRetry`, I think `Background` here is okay, but remember to leave a comment to tell the reason